### PR TITLE
JDK-8314246: javax/swing/JToolBar/4529206/bug4529206.java fails intermittently on Linux

### PR DESCRIPTION
**Pull request by honkar-jdk**
JToolBar test - bug4529206.java was failing intermittently on Linux. It was due to timing issue. Added appropriate delay to fix the issue.

CI Testing passes on all platforms.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8314246](https://bugs.openjdk.org/browse/JDK-8314246): javax/swing/JToolBar/4529206/bug4529206.java fails intermittently on Linux (**Bug** - P4)

### Reviewers
 * [Damon Nguyen](https://openjdk.org/census#dnguyen) (@DamonGuy - Committer) ⚠️ Review applies to [3b4ae1be](https://git.openjdk.org/jdk/pull/15316/files/3b4ae1be282cc58ca4d905440a930f0e73e9b161)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15316/head:pull/15316` \
`$ git checkout pull/15316`

Update a local copy of the PR: \
`$ git checkout pull/15316` \
`$ git pull https://git.openjdk.org/jdk.git pull/15316/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15316`

View PR using the GUI difftool: \
`$ git pr show -t 15316`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15316.diff">https://git.openjdk.org/jdk/pull/15316.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15316#issuecomment-1681090526)

### CODE REVIEW

File: test/jdk/javax/swing/JToolBar/4529206/bug4529206.java [hunk 1]
```diff
@@ -77,32 +71,25 @@ private static void makeToolbarFloat() {
...
     public static void main(String[] args) throws Exception {
         try {
-            SwingUtilities.invokeAndWait(new Runnable() {
-                public void run() {
-                    test();
-                }
-            });
+            SwingUtilities.invokeAndWait(() -> test());
             Robot robot = new Robot();
             robot.waitForIdle();
```
Comment (DamonGuy) [inline test/jdk/javax/swing/JToolBar/4529206/bug4529206.java:82]: Changes look good. Did you try using using setAutoWaitForIdle?

**Review by DamonGuy — APPROVED** [review]

File: test/jdk/javax/swing/JToolBar/4529206/bug4529206.java (hunk 1 above)
Comment (honkar-jdk) [reply test/jdk/javax/swing/JToolBar/4529206/bug4529206.java]: Oh, I didn't notice that `setAutoWaitForIdle()` was not used the first time. I have changed it. I think adding a small amount of delay after `makeToolbarFloat()` is good in addition to `setAutoWaitForIdle()` to stabilize the test on slower systems.

**Review by mrserb — APPROVED** [review]

File: test/jdk/javax/swing/JToolBar/4529206/bug4529206.java (hunk 1 above)
Comment (aivanov-jdk) [reply test/jdk/javax/swing/JToolBar/4529206/bug4529206.java]: > Oh, I didn't notice that `setAutoWaitForIdle()` was not used the first time. I have changed it. I think adding a small amount of delay after `makeToolbarFloat()` is good in addition to `setAutoWaitForIdle()` to stabilize the test on slower systems.

I think this was a bad decision: now the test **never** calls `waitForIdle()`. But it should do it… for stability… after it takes any action.

This test does not use `Robot` to send any events, but `robot.setAutoWaitForIdle(true)` affects only *mouse and keyboard events*: [`Robot.autoWaitForIdle()`](https://github.com/openjdk/jdk/blob/6fc35142315f1616fa35e415005c9483939c6920/src/java.desktop/share/classes/java/awt/Robot.java#L688) is called from `afterEvent`, just like `autoDelay`:

https://github.com/openjdk/jdk/blob/6fc35142315f1616fa35e415005c9483939c6920/src/java.desktop/share/classes/java/awt/Robot.java#L662-L665

I submitted [JDK-8318423](https://bugs.openjdk.org/browse/JDK-8318423): Add explicit robot.waitForIdle to JToolBar/4529206/bug4529206.java

**Pull request closed without merging**